File: nexus/strings.py
from getpass import getuser
import os
from os.path import *

# Environment
username = os.environ.get('SUDO_USER', os.environ.get('USERNAME'))
if username == None:
  username = getuser()

# Paths
home_dir = expanduser(f'~{username}')
nexus_dir = home_dir+'/.nexus'
dockerfiles_dir = nexus_dir+'/dockerfiles'
buildfiles_dir = nexus_dir+'/data'
image_cache_dir = nexus_dir+'/imagecache'

# Messages
program_usage_msg = \
"""Usage: nexus [command]
- nexus init
    Initializes the directories required to use nexus (~/.nexus/*)
- nexus enter [image]
    Enters an image defined in ~/.nexus/dockerfiles/
- nexus rebuild [image]
    Forces a rebuild of an image
    Use `--no-cache` to not use cache when building
- nexus clean [mode]
    Modes:
      cache - cleans nexus image cache
      env   - cleans nexus docker environment
      all   - cleans both the image cache and docker environment"""

File: strings.go
package main

import (
    "os"
)

const helpMessage =
`Usage: nexus [command] [params]

- nexus clean [option]
    - cache
        Clean built image cache located at ~/.cache/nexus/imagecache/
    - environment / env
        Clean Nexus environment container
    - all
        Clean both

- nexus enter [dockerfile] [command]
    Builds and/or enters the Dockerfile located at ~/.config/nexus/dockerfiles/<dockerfile>
    If [command] is specified, that command will be run when entering the container, otherwise
    '/bin/sh' will run by default

- nexus rebuild [dockerfile] [--no-cache]
    Builds/rebuilds the Dockerfile located at ~/.config/nexus/dockerfiles/<dockerfile>
    If the --no-cache flag is specified, the image is built without the use of Docker cache

- nexus start
    Starts the nexus docker environment, or creates it if it doesn't exist`

const commonDockerFlags = "--privileged --device=/dev/dri --device=/dev/fuse --env=DISPLAY --net=host"

var homeDir, configDir, dockerfilesDir, buildfilesDir, cacheDir, imagecacheDir string

func initStrings() {
    homeDir, _ = os.UserHomeDir()
    configDir = homeDir + "/.config/nexus"
    dockerfilesDir = configDir + "/dockerfiles"
    buildfilesDir = configDir + "/buildfiles"
    cacheDir = homeDir + "/.cache/nexus"
    imagecacheDir = cacheDir + "/imagecache"
}
